### PR TITLE
Add logging for batch header hash for batcher-node

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
To make it easier to correlate batcher-node communication on a per batch basis

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: disperser/batcher/grpc/dispatcher.go [hunk 1]
```diff
@@ -79,13 +83,15 @@ func (c *dispatcher) sendAllChunks(ctx context.Context, state *core.IndexedOpera
 					Signature: nil,
 					Operator:  id,
 				}
+				c.logger.Warn("Failed to send chunks to operator", "batchHeaderHash", batchHeaderHash, "operator", op.Socket, "err", err)
```
Comment (ian-shim) [inline disperser/batcher/grpc/dispatcher.go:86]: can we also log operator IDs here and below?

**Review by ian-shim — APPROVED** [review]
I don't have a strong opinion but this may result in a lot of extra lines of logs.
What do you think about passing the batch header hash in the signer message and logging it as part of logs [here](https://github.com/Layr-Labs/eigenda/blob/master/core/aggregation.go#L130) and [here](https://github.com/Layr-Labs/eigenda/blob/master/core/aggregation.go#L173)?

**Comment by jianoaix** [comment]
> here

Sounds good to me

**Comment by jianoaix** [comment]
Tested:

`2024-05-08 13:22:02.220 | {"time":"2024-05-08T20:22:02.220421429Z","level":"WARN","source":{"function":"github.com/Layr-Labs/eigenda/core.(*StdSignatureAggregator).AggregateSignatures","file":"/app/core/aggregation.go","line":131},"msg":"error returned from messageChan","component":"SignatureAggregator","operatorID":"39638352822b2eecf33f2ea0e6893a576edbcc202b4925c3ab2875677945bf58","operatorAddress":"0xad9770d6b5514724c7b766f087bea8a784038cbe","socket":"18.208.146.49:32007;32008","batchHeaderHash":[147,252,95,99,206,177,111,125,109,84,0,119,245,234,89,206,143,59,28,166,14,208,146,106,212,88,205,139,38,154,149,48],"err":"rpc error: code = DeadlineExceeded desc = context deadline exceeded"} `

**Pull request merged**